src/helpers/git_helpers/branch_configuration_helpers.sh

```diff
@@ -94,89 +123,72 @@ function echo_update_child_branches {
 
 # Makes sure that we know all the parent branches
 # Asks the user if necessary
-# Aborts the script if not all branches become known.
+#
+# branches - space seperated list of branches
 function ensure_knows_parent_branches {
-  local current_branch=$1
-
-  if [ "$(knows_all_ancestor_branches "$current_branch")" = true ]; then
-    return
-  fi
-
-  # Here we don't have the ancestors list --> make sure we know all ancestors, then recompile it from all ancestors
+  local branches=$1
+  local branch
+  local child
   local parent
+  local user_input
+  local numerical_regex='^[0-9]+$'
+  local header_shown=false
+
+  for branch in $branches; do
+    child=$branch
+    if [ "$(knows_all_ancestor_branches "$child")" = true ]; then
+      continue
+    fi
 
-  while [ "$current_branch" != "$MAIN_BRANCH_NAME" ]; do
-    if [ "$(knows_parent_branch "$current_branch")" = true ]; then
-      parent=$(parent_branch "$current_branch")
-    else
-      # here we don't know the parent of the current branch -> ask the user
-      echo
-      local branches=$(git for-each-ref --sort=-committerdate refs/heads/ --format='%(refname:short)')
-
-      function print_branch {
-        local number=$1
-        local branch_name=$2
-
-        output_style_bold
-        printf "%3s: " "$number"
-        output_style_reset
-        echo "$branch_name"
-      }
-
-      echo
-      echo "Feature branches can be branched directly off "
-      echo "$MAIN_BRANCH_NAME or from other feature branches."
-      echo
-      echo "The former allows to develop and ship features completely independent of each other."
-      echo "The latter allows to build on top of currently unshipped features."
-      echo
-
-      local branch_numbers
-      print_branch 1 "$MAIN_BRANCH_NAME"
-      branch_numbers[1]=$MAIN_BRANCH_NAME
-      i=2
-      for branch in $branches; do
-        if [ "$branch" != "$current_branch" ] && [ "$branch" != "$MAIN_BRANCH_NAME" ]; then
-          branch_numbers[i]=$branch
-          print_branch $i "$branch"
-          i=$(( i + 1 ))
+    while [ "$child" != "$MAIN_BRANCH_NAME" ]; do
+      if [ "$(knows_parent_branch "$child")" = true ]; then
+        parent=$(parent_branch "$child")
+      else
+        if [ "$header_shown" = false ]; then
```

Comment:
How do you feel about naming this variable `description_shown`? I think "header" is a bit ambiguous here. There are many different headers that commands can show.